app/controllers/comments_controller.rb

```diff
@@ -202,6 +203,61 @@ def preview
     end
   end
 
+  ## TODO: Service-orient this method - SubscriptionCreator.call(...)
+  # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
```

Comment:
So, I think I agree with Rubocop here: there's too much complexity in this action. It looks like that complexity isn't inherent to the complexity of the work and I think we can smooth some of it out prior to merging. I would think moving the logic here into a service-object would be the biggest win here for keeping this complexity manageable.

Also, having been particularly frustrated by the way our `reactions` end-points operate as "complicated toggles", I think we should approach this with separate `POST` and `DELETE` end-points. That _alone_ would help with readability and tone down the complexity. Since this action shares very little with the other actions in this controller, I think it might help to split it out into a new controller, that way it can use default `CRUD`-type actions, as well.